pilot/pkg/serviceregistry/kube/controller/ambient/policies.go

```diff
@@ -84,6 +91,46 @@ func PolicyCollections(
 		AuthzDerivedPolicies,
 		PeerAuthDerivedPolicies,
 		DefaultPolicy.AsCollection(),
+		ImplicitWaypointPolicies,
 	}, krt.WithName("Policies"))
 	return AuthzDerivedPolicies, Policies
 }
+
+func implicitWaypointPolicyName(waypoint *Waypoint) string {
+	if !features.DefaultAllowFromWaypoint || waypoint == nil || len(waypoint.ServiceAccounts) == 0 {
+		return ""
+	}
+	// use '_' character since those are illegal in k8s names
+	return "istio_allow_waypoint_" + waypoint.Namespace + "_" + waypoint.Name
+}
+
+func implicitWaypointPolicy(ctx krt.HandlerContext, waypoint Waypoint) *model.WorkloadAuthorization {
+	if !features.DefaultAllowFromWaypoint || len(waypoint.ServiceAccounts) == 0 {
+		return nil
+	}
+	return &model.WorkloadAuthorization{
```

Comment:
One problem with this I cannot create a more restrictive policy like "allow waypoint on port 80". Since the allows are ORs

Reply:
Hmmm. So if policies > 0:

1. Add allow source == waypoint to each existing policy
2. Do nothing and assume user policy will handle the waypoint

1 seems a little too magic and 2 seems like a footgun.

Reply:
if we leave this as is the feature makes it so 

* Waypoint can do whatever it want. 
* You can add a DENY waypoint for some reason to stop traffic from it.
*  You can also add ALLOW other things to make it so it's no longer waypoint-exclusive.

Reply:
I agree with steven; you can create a DENY policy with not matches

Reply:
Keep in mind we don't have to fit everything in policies. 

We can have explicit code in ztunnel to allow traffic from some service account (using a naming pattern), without messing with the rules or pretending that we know someone using a service account is a waypoint ( that's not enforceable AFAIK ).

We still have the pretty insecure 'secure naming' where we guess what service accounts to allow based on what workloads with some labels run as.  Is this the kind of 'implicit' we are talking about ? Is there a doc with the security properties of the system that someone reviewing the security posture can read instead of guessing from the code ( and we can use to validate the code security ) ?

Reply:
This is also  my suggestions https://github.com/istio/istio/pull/50475#pullrequestreview-2008037649
I felt we donot need to do any change in istiod.